lib/iiif/v3/presentation/nav_place.rb

```diff
@@ -0,0 +1,98 @@
+require 'geo/coord'
+
+module IIIF
+  module V3
+    module Presentation
+      class NavPlace < IIIF::V3::AbstractResource
+        Rect = Struct.new(:coord1, :coord2)
+
+        def initialize(coordinate_texts:, base_uri:)
+          @coordinate_texts = coordinate_texts
+          @base_uri = base_uri
+        end
+
+        def build
+          return if coordinates.nil? || coordinates.empty?
+
+          {
+            id: "#{base_uri}/feature-collection/1",
+            type: 'FeatureCollection',
+            features:
+          }
+        end
+
+        private
+
+        attr_reader :coordinate_texts, :base_uri
+
+        def coordinates
+          @coordinates ||= coordinate_texts.map do |coordinate_text|
+            coordinate_parts = coordinate_text.split(%r{ ?--|/})
+            case coordinate_parts.length
+            when 2
+              coord_for(coordinate_parts[0], coordinate_parts[1])
+            when 4
+              rect_for(coordinate_parts)
+            end
+          end.compact
+        end
+
+        COORD_REGEX = /(?<hemisphere>[NSEW]) (?<degrees>\d+)[°⁰*] ?(?<minutes>\d+)?[ʹ']? ?(?<seconds>\d+)?[ʺ"]?/
+
+        def coord_for(long_str, lat_str)
+          long_matcher = long_str.match(COORD_REGEX)
+          lat_matcher = lat_str.match(COORD_REGEX)
+          return unless long_matcher && lat_matcher
```

Comment:
Do we want to silently fail here, or would an argument exception make it clear that what you are doing is not valid?

Reply:
Not sure if I am the correct person to answer this, as this was just a refactor PR to move existing code in PURL to here....but my guess is that by silently failing, we prevent the method from blowing up if there is geo data but it is not formatted correctly.  If we throw an exception, in these cases I'd think we would hit errors in PURL until the data is remediated?

Reply:
I think you ought to check if you have usable data before calling this method.  I think this should let you know when you give it the wrong data. Otherwise it becomes a mystery you have to track down.

Reply:
Suggestion:

1. Add a new method to this class to validate coordinates -- it returns true/false
2. Caller (currently only PURL), calls this first and then only calls `build` if valid, otherwise noop (or if you really want to raise in PURL, we can do that too, but I will note that is not what is happening in production at the moment and for this refactor work, I'd suggest preserving functionality as is).
3. `build` method raises an exception if invalid coordinates are sent in anyway.

Is that what you are thinking

Reply:
I'm only thinking of 3 for this gem.  You, as the user of the gem can determine how you want to ensure what you're passing in is "good". That could be validating the data before you store it, or filtering out the data you have before you call this method.

Reply:
Did both 1 & 3 (added a `.valid?` method and used it to determine if we raise an exception on `.build?`).  PURL can use this `.valid?` method to keep current functionality as is.

Reply:
Should the validate method be checking the COORD_REGEX though?

Reply:
Nevermind, I see it does that.